### ISSUE TITLE
"npm update tap --depth=1" fails with confusing error

### ISSUE THREAD

**Issue by aredridel** [npm/npm#9473]
```
npm ERR! Darwin 15.0.0
npm ERR! argv "node" "/usr/local/bin/npm" "update" "tap" "--depth=1"
npm ERR! node v0.12.7
npm ERR! npm  v3.3.1

npm ERR! Cannot read property 'filter' of undefined
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/aredridel/Projects/npm/newww/npm-debug.log
```

debug log:

```
0 info it worked if it ends with ok
1 verbose cli [ 'node', '/usr/local/bin/npm', 'update', 'tap', '--depth=1' ]
2 info using npm@3.3.1
3 info using node@v0.12.7
4 verbose node symlink /usr/local/bin/node
5 silly mapToRegistry name tap
6 silly mapToRegistry using default registry
7 silly mapToRegistry registry https://registry.npmjs.org/
8 silly mapToRegistry uri https://registry.npmjs.org/tap
9 verbose stack TypeError: Cannot read property 'filter' of undefined
9 verbose stack     at outdated_ (/usr/local/lib/node_modules/npm/lib/outdated.js:212:27)
9 verbose stack     at skip (/usr/local/lib/node_modules/npm/lib/outdated.js:318:5)
9 verbose stack     at shouldUpdate (/usr/local/lib/node_modules/npm/lib/outdated.js:333:55)
9 verbose stack     at /usr/local/lib/node_modules/npm/lib/outdated.js:301:23
9 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:52:35
9 verbose stack     at Array.forEach (native)
9 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:52:11
9 verbose stack     at Array.forEach (native)
9 verbose stack     at asyncMap (/usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:51:8)
9 verbose stack     at outdated_ (/usr/local/lib/node_modules/npm/lib/outdated.js:294:3)
10 verbose cwd /Users/aredridel/Projects/npm/newww
11 error Darwin 15.0.0
12 error argv "node" "/usr/local/bin/npm" "update" "tap" "--depth=1"
13 error node v0.12.7
14 error npm  v3.3.1
15 error Cannot read property 'filter' of undefined
16 error If you need help, you may report this error at:
16 error     <https://github.com/npm/npm/issues>
17 verbose exit [ 1, true ]
```

**Comment by othiym23** [comment]
This is one for @iarna, but it also looks pretty easy to fix. Since it's not on one of the primary use cases for the CLI, I'm not going to raise it to being a blocker even though it's pretty clearly a regression. Let me know if you disagree, @aredridel.

**Comment by aredridel** [comment]
Not a problem! Just stubbed my toe on it.

**Comment by zkat** [comment]
I can't really reproduce it. Whatever was causing `tree.children` to not be an array or array-like or whatever it was may have been fixed, or my clone of `newww` can't really reproduce it anymore.

**Comment by iarna** [comment]
I'm going to suggest that this has probably been fixed since `npm@3.3.1` from nearly a year ago. I'm gonna close this, but if anyone runs into something like this again, please open a new issue.

**Comment by dbuezas** [comment]
Happening again. In my case it is a scoped (and private) package and it has a dep which is also a scoped (and private) package.
npm --version --> scoped (and private) package 5.4.1
node --version --> v7.10.1

tree.children is undefined
